internal/envelope/envelope.go

```diff
@@ -23,3 +43,29 @@ func GetEnvelopeMediaType(sigFormat string) (string, error) {
 	}
 	return "", fmt.Errorf("signature format %q not supported", sigFormat)
 }
+
+// ValidatePayloadContentType validates signature payload's content type.
+func ValidatePayloadContentType(payload *signature.Payload) error {
+	switch payload.ContentType {
+	case MediaTypePayloadV1:
+		return nil
+	default:
+		return fmt.Errorf("payload content type %q not supported", payload.ContentType)
+	}
+}
+
+// DescriptorFromPayload parses a signature payload and returns the descriptor
+// that was signed. Note: the descriptor was signed but may not be trusted
+func DescriptorFromSignaturePayload(payload *signature.Payload) (*ocispec.Descriptor, error) {
+	if payload == nil {
+		return nil, errors.New("empty payload")
+	}
+
+	var parsedPayload Payload
+	err := json.Unmarshal(payload.Content, &parsedPayload)
+	if err != nil {
+		return nil, errors.New("failed to unmarshall the payload content to Payload")
+	}
+
+	return &parsedPayload.TargetArtifact, nil
+}
```

Comment:
IMO, from code reusability perspective, we could call `ValidatePayloadContentType` inside `DescriptorFromSignaturePayload`. Otherwise, using `DescriptorFromSignaturePayload` would require calling `ValidatePayloadContentType` beforehand.